src/clr-angular/forms/datepicker/date-input.ts

```diff
@@ -161,11 +123,31 @@ export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implement
             // the ngOnInit hook.
             this.dateValueOnInitialLoad = value;
         } else {
-            this.processDate(value);
+            this.processUserDateObject(value);
         }
     }
 
-    @Output("clrDateChange") _dateUpdated: EventEmitter<Date> = new EventEmitter<Date>(false);
+    /**
+     * Processes a date object to check if its valid or not.
+     */
+    processUserDateObject(value: Date) {
+        if (this._dateIOService) {
+            // The date object is converted back to string because in Javascript you can create a date object
+            // like this: new Date("Test"). This is a date object but it is invalid. Converting the date object
+            // that the user passed helps us to verify the validity of the date object.
+            const dateStr: string = this._dateIOService.toLocaleDisplayFormatString(value);
+            const date: Date = this._dateIOService.isValidInput(dateStr);
+            if (date) {
+                const dayModel: DayModel = new DayModel(date.getFullYear(), date.getMonth(), date.getDate());
+                if (!dayModel.isEqual(this._dateNavigationService.selectedDay)) {
+                    this._dateNavigationService.selectedDay = dayModel;
+                    this.writeDateStrToInputField(dateStr);
+                }
+            } else {
+                this._dateNavigationService.selectedDay = null;
```

Comment:
Shouldn't you `this.writeDateStrToInputField("");`?

Reply:
I don't think we should. If the user enters `02/30/2015` (Incorrect Feb date) and focuses out, we don't want to override what he entered when the user focuses out of the input. We just know that the `selectedDay` is `null` and open the Calendar on today's date.

Reply:
please let me know if my assumption here is incorrect.

Reply:
I believe we wanted to take this approach until we have masked inputs, no?

Reply:
It's the approach we want for masked inputs, but only because for them there is feedback that the date is wrong. In your case we don't have any error showing, so if I type `02/30/2015` and move on to the next state, it looks like everything is fine and I entered correctly.

So you are right in that I still stand behind the "don't erase user input", but I'm not sure what the behavior should be here. We can ship like this, but please open an issue immediately about this.

Reply:
@youdz I believe what we wanted to do is to integrate the new validation which will come with the new forms into the datepicker when this happens. I will check with UX what to do until then. I will open the ticket as soon as I merge this into master.